clang/lib/Tooling/Tooling.cpp

```diff
@@ -722,6 +722,13 @@ int ClangTool::proccessFiles(llvm::StringRef File,bool &ProcessingFailed,
         return -29 /*MigrationErrorCannotAccessDirInDatabase*/;
       }
 
+      if (CompileCommand.Heuristic != "") {
```

Comment:
For file not in compilation database,  dpct should mgirate without Heurstic rules auto generated.